### DISCUSSION TITLE
`yarn build` should run at build time

### DISCUSSION THREAD

**Discussion by zuc** [outline/outline#1319] [General] [answered]
Hi there,
is there any specific reason while the project's Dockerfile does this?

```Dockerfile
CMD yarn build && yarn start
```

It feels really weird as this implies having a bootstrap time of ~50 sec. for each spawned container (I'm running outline on top of a Kubernetes cluster).

IMHO, it should instead do something like:

```Dockerfile
RUN yarn build
CMD yarn start
```

so that `node_modules` is built while baking the image.
WDYT?

**Comment by tommoor** [comment]
I believe the main reason that's left is that the build process includes some environment variables:

https://github.com/outline/outline/blob/develop/webpack.config.prod.js#L34-L40

**Reply by zuc** [reply]
Yes, I got to the same conclusion. Still, I think this is an anti-pattern.
I wonder whether there's interest in reconsidering this.

**Reply by mavrick** [reply]
i haven't used Koa before, but would it be possible to pass these env vars using the `render` method instead of `koa-sendfile` ? then just use `window` or localstorage to access them?

i've had to clone the repo and move the `yarn build` into my cloudbuild script instead as I am also deploying to k8s and didn't want to assign a lot of memory for the pod.

**Reply by tommoor** [reply]
Yes, absolutely – that's the way out of this – inject a script tag with the variables.

**Reply by tommoor** [reply]
PR here: https://github.com/outline/outline/pull/1353

**Comment by tommoor** [comment, marked as answer]
As of [v0.45.0](https://github.com/outline/outline/releases/tag/v0.45.0) there is no longer a build step on docker startup
